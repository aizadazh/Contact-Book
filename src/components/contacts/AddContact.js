import React, { Component } from "react";
import propTypes from "prop-types";
import TextInputGroup from "../layout/TextInputGroup";
import { connect } from "react-redux";
import { addContact } from "../../actions/ContactAction";

class AddContact extends Component {
    state = {
        name: "",
        address:"",
        phone: "",
        errors: {},
    };
    onSubmit = e => {
        e.preventDefault();

        const { name, address, phone } = this.state;
        if (name === "") {
            this.setState({errors: {name: "Пожалуйста, введите имя"} });
            return;
        }
        if (phone ==="") {
            this.setState({errors: {phone: "Пожалуйста, введите номер телефона"} });
            return;
        }
        if (address ==="") {
            this.setState({errors: {address: "Пожалуйста, введите адрес"} });
            return;
        }
        const newContact = {
            name,
            address,
            phone,
        };
        this.props.addContact(newContact);

        this.setState = ({
            name: "",
            address: "",
            phone: "",
            errors: {},
        });

        this.props.history.push("/");
    };
    onChange = e => this.setState({ [e.target.name]: e.target.value });

    render() {
        const { name, address, phone, errors } = this.state;  

        return (
            <div className="card mb-3">
                <div className="card-header">Добавить контакт</div>
                    <div className="card-body">
                        <form onSubmit={this.onSubmit}>
                            <TextInputGroup
                                label="ФИО"
                                name="name"
                                placeholder="Введите имя"
                                value={name}
                                onChange={this.onChange}
                                error={errors.name}
                            />
                            <TextInputGroup
                                label="Адрес"
                                name="address"
                                type="address"
                                placeholder="Введите адрес"
                                value={address}
                                onChange={this.onChange}
                                error={errors.address}
                            />
                            <TextInputGroup
                                label="Номер телефона"
                                name="phone"
                                placeholder="Введите номер телефона"
                                value={phone}
                                onChange={this.onChange}
                                error={errors.phone}
                            />
                            <input
                                type="submit"
                                value="Добавить контакт"
                                className="btn btn-danger btn-block"
                            />
                        </form>
                    </div>
                </div>   
        );
    }
}
AddContact.propTypes = {
    addContact: propTypes.func.isRequired,
    };
export default connect(
    null,
    { addContact },
    )(AddContact);


