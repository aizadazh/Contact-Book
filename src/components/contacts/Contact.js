import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import propTypes from "prop-types";
import { deleteContact } from "../../actions/ContactAction";


export class Contact extends Component {
    state = {
        showContactInfo: false,
    };
    onDeleteClick  = id => {
        this.props.deleteContact(id);
    };
    render(){
        const { id, name, address, phone } = this.props.contact;
        const { showContactInfo} = this.state;
        return(
            <div>
                <h4>
                    {name}{" "}
                    <i 
                        onClick = {() =>
                            this.setState({
                                showContactInfo: !this.state.showContactInfo,
                            })
                        }
                        className = "fas fa-sort-down"
                        style = {{cursor: "pointer"}}
                        />
                    <i 
                        className = "fas fa-times"
                        style = {{cursor: "pointer", float:"right", color: "red"}} 
                        onClick = {this.onDeleteClick.bind(this, id)}
                    />
                    <Link to ={`contact/edit/${id}`}>
                        <i
                            className = "fas fa-pencil-alt"
                            style={{
                                cursor: "pointer",
                                float: "right",
                                color: "grey",                               
                                marginRight: "1rem",
                            }}
                        />
                    </Link>                    
                </h4>
                    {showContactInfo ? (
                        <ul className="list-group">
                            <li className="list-group-item">Address: {address}</li>
                            <li className="list-group-item">Phone: {phone}</li>
                        </ul>
                    ) : null }
            </div>
        );
    }
}
Contact.propTypes = {
    contact: propTypes.object.isRequired,
    deleteContact: propTypes.func.isRequired,
};
export default connect(
    null, 
    { deleteContact },    
)(Contact);
