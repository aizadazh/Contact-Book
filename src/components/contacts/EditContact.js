import React, { Component } from "react";
import TextInputGroup from "../layout/TextInputGroup";
import { connect } from "react-redux";
import propTypes from "prop-types";
import { getContact, editContact } from "../../actions/ContactAction";

class EditContact extends Component {
  state = {
    name: "",
    address: "",
    phone: "",
    errors: {},
  };

  componentDidMount() {
    const { id } = this.props.match.params;
    this.props.getContact(id);
  }
  componentWillReceiveProps(nextProps, nextState) {
    const { name, address, phone } = nextProps.contact;
    this.setState({ name, address, phone });
  }

  onSubmit = e => {
    e.preventDefault();

    const { name, address, phone } = this.state;
    
    if (name === "") {
      this.setState({ errors: { name: "Введите имя" } });
      return;
    }

    if (address === "") {
      this.setState({ errors: { address: "Введите адрес" } });
      return;
    }

    if (phone === "") {
      this.setState({ errors: { phone: "Введите номер телефона" } });
      return;
    }

    const { id } = this.props.match.params;

    const editedContact = {
      id,
      name,
      address,
      phone,
    };
    this.props.editContact(editedContact);
    
    this.setState({
      name: "",
      address: "",
      phone: "",
      errors: {},
    });

    this.props.history.push("/");
  };

  onChange = e => this.setState({ [e.target.name]: e.target.value });

  render() {
    const { name, address, phone, errors } = this.state;

    return (
      <div className="card mb-3">
        <div className="card-header">Редактировать контакт</div>
        <div className="card-body">
          <form onSubmit={this.onSubmit}>
            <TextInputGroup
              label="ФИО"
              name="name"
              placeholder="Введите имя"
              value={name}
              onChange={this.onChange}
              error={errors.name}
            />
            <TextInputGroup
              label="Адрес"
              name="address"
              type="address"
              placeholder="Введите адрес"
              value={address}
              onChange={this.onChange}
              error={errors.address}
            />
            <TextInputGroup
              label="Номер телефона"
              name="phone"
              placeholder="Введите номер телефона"
              value={phone}
              onChange={this.onChange}
              error={errors.phone}
            />
            <input
              type="submit"
              value="Редактировать контакт"
              className="btn btn-dark btn-block"
            />
          </form>
        </div>
      </div>
    );
  }
}
EditContact.propTypes = {
  contact: propTypes.string.isRequired,
  getContact: propTypes.func.isRequired,
};
const mapStateToProps = state => ({
  contact: state.contact.contact,
});
export default connect(
  mapStateToProps,
  { getContact, editContact },
)(EditContact);