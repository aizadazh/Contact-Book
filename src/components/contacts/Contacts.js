import React, { Component, Fragment } from "react";
import Contact from "./Contact";
import { connect } from "react-redux";
import propTypes from "prop-types";
import { getContacts } from "../../actions/ContactAction";

class Contacts extends Component {
  componentDidMount() {
    this.props.getContacts();
  }
  render() {    
    const { contacts } = this.props;
    return (
      <Fragment>
        <h1 className="display-5 mb-2">
          <span className="text-danger">Список контактов</span>
        </h1>
        {contacts.map(contact => (
          <Contact key={contact.id} contact={contact} />
        ))}
      </Fragment>
    );
  }
}
Contacts.propTypes = {
  contacts: propTypes.array.isRequired,
  getContacts: propTypes.func.isRequired,
};

const mapStateToProps = state => ({
  contacts: state.contact.contacts,
});

export default connect(
  mapStateToProps,
  { getContacts },
)(Contacts);

