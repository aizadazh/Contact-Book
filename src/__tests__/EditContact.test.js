import React from "react";
import { shallow } from "enzyme";
import { EditContact } from "../components/contacts/EditContact";

function setup() {
    const props = {
        contact: "Aizada Zhumalieva",
        getContact: jest.fn(),
        match: { params: "params" }
    }
    const wrapper = shallow(<EditContact {...props} />)
    return {
        props,
        wrapper
    }
}
describe("Testing EditContact component", ()=> {
    it('renders correctly', ()=> {
        const { wrapper } = setup()
        expect(wrapper.find('TextGroupInput')).toBeDefined()
        expect(wrapper).toMatchSnapshot();
    })
})