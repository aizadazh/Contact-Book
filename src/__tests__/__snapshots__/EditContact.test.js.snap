// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing EditContact component renders correctly 1`] = `ShallowWrapper {}`;
